interface/wx/clntdata.h

```diff
@@ -59,6 +59,50 @@ class wxClientDataContainer
 
 
 
+/**
+    @class wxSharedClientDataContainer
+
+    This class is a replacement for @ref wxClientDataContainer, and unlike
+    wxClientDataContainer the wxSharedClientDataContainer client data is
+    possible to copy (as a shared ptr) when instances of it are cloned.
+    Like wxClientDataContainer, wxSharedClientDataContainer is a mixin
+    that provides storage and management of "client data.". The client data
+    is reference counted and managed by the container.
+
+    @note If your class has a Clone function and needs to store client data,
+          use wxSharedClientDataContainer and not wxClientDataContainer!
+
+    @library{wxbase}
+    @category{containers}
+
+    @see wxClientDataContainer, wxClientData
+*/
```

Comment:
We're going to need this here:
```suggestion
    @since 3.1.7
*/
```